StackExchange.Redis/StackExchange/Redis/Interfaces/IDatabase.cs

```diff
@@ -1389,6 +1389,250 @@ public interface IDatabase : IRedis, IDatabaseAsync
+
+        /// <summary>
+        /// Read a stream using the given range of IDs.
+        /// </summary>
+        /// <param name="key">The key of the stream.</param>
+        /// <param name="minId">The minimum ID from which to read the stream of pending messages. Pass "-" or <see cref="StreamConstants.ReadMinValue"/> to read from the beginning of the stream" /></param>
+        /// <param name="maxId">The maximum ID to read to within the stream of pending messages. Pass "+" or <see cref="StreamConstants.ReadMaxValue"/> to read to the end of the stream.</param>
+        /// <param name="count">The maximum number of messages to return.</param>
+        /// <param name="messageOrder">The order of the messages. <see cref="Order.Ascending"/> will execute XRANGE and <see cref="Order.Descending"/> wil execute XREVRANGE.</param>
+        /// <param name="flags">The flags to use for this operation.</param>
+        /// <returns>Returns an instance of <see cref="RedisStreamEntry"/> for each message returned.</returns>
+        /// <remarks>https://redis.io/commands/xrange</remarks>
+        RedisStreamEntry[] StreamRange(RedisKey key, RedisValue minId, RedisValue maxId, int? count = null, Order messageOrder = Order.Ascending, CommandFlags flags = CommandFlags.None);
+
+        /// <summary>
+        /// Read from a single stream.
+        /// </summary>
+        /// <param name="key">The key of the stream.</param>
+        /// <param name="afterId">The position from within the stream to begin reading.</param>
+        /// <param name="count">The maximum number of messages to return.</param>
+        /// <param name="flags">The flags to use for this operation.</param>
+        /// <returns>Returns an instance of <see cref="RedisStreamEntry"/> for each message returned.</returns>
+        /// <remarks>Equivalent of calling XREAD COUNT num STREAMS key id.</remarks>
+        /// <remarks>https://redis.io/commands/xread</remarks>
+        RedisStreamEntry[] StreamRead(RedisKey key, RedisValue afterId, int? count = null, CommandFlags flags = CommandFlags.None);
+
+        /// <summary>
+        /// Read from multiple streams.
+        /// </summary>
+        /// <param name="streamAndIdPairs">The list of streams and the ID from which to begin reading for each stream.</param>
+        /// <param name="countPerStream">The maximum number of messages to return from each stream.</param>
+        /// <param name="flags">The flags to use for this operation.</param>
+        /// <returns>An instance of <see cref="RedisStream"/> for each stream.</returns>
+        /// <remarks>Equivalent of calling XREAD COUNT num STREAMS key1 key2 id1 id2.</remarks>
+        /// <remarks>https://redis.io/commands/xread</remarks>
+        RedisStream[] StreamRead(IList<KeyValuePair<RedisKey, RedisValue>> streamAndIdPairs, int? countPerStream = null, CommandFlags flags = CommandFlags.None);
```

Comment:
@mgravell thoughts here? We don't have this type of input type (AFAIK) anywhere else in the API, the closest equivalent is `SortedSetEntry[]` for adds, which incurs the array overhead you'd likely want to avoid in streams in general...suggestions?

Reply:
I’ve always thought this one looked clunky. An input type tailored for this method would look a lot cleaner.

Something like this?
StreamRead(StreamIdPair[] streamIdPairs, int? countPerStream, ...)